java/jdbc/src/main/java/io/vitess/jdbc/VitessPreparedStatement.java

```diff
@@ -752,8 +752,14 @@ public void setBlob(int parameterIndex, Blob x) throws SQLException {
     }
 
     public void setClob(int parameterIndex, Clob x) throws SQLException {
-        throw new SQLFeatureNotSupportedException(
-            Constants.SQLExceptionMessages.SQL_FEATURE_NOT_SUPPORTED);
+        checkOpen();
+        if (x.length() > Integer.MAX_VALUE) {
+            throw new SQLFeatureNotSupportedException(
```

Comment:
Theoretically we could just send any >2gb value as a ByteString but in practice it has to be streamed. The current Vitess protobuffers don't support streaming query parameters or results. Supporting that would be a massive restructure of the API.

I think we should punt on that for now.